site/content/en/contributions/design/backend.md

```diff
@@ -0,0 +1,159 @@
+- Cluster-external endpoints, which are currently second-class citizens in Gateway-API 
+  (supported using [Services and FQDN endpoints][]).
+- Host-local endpoints, such as sidecars or daemons that listen on [unix domain sockets][] or envoy [internal listeners][], 
+  that cannot be represented by a K8s service at all.
+
+Several projects currently support backends that are not registered in the infrastructure-specific service registry. 
+- K8s Ingress: [Resource Backends][]
+- Istio: [Service Entry][]
+- Gloo Edge: [Upstream][]
+- Consul: [External Services][]
+
+## Goals
+* Add an API definition to hold settings for configuring Unix Domain Socket, FQDN and IP.
+* Determine which resources may reference the new backend resource.
+* Determine which existing Gateway-API and Envoy Gateway policies may attach to the new backend resource. 
+
+## Non Goals
+* Support specific backend types, such as S3 Bucket, Redis, AMQP, InfluxDB, etc.  
+
+## Implementation
+
+The `Backend` resource is an implementation-specific Gateway-API [BackendObjectReference Extension][]. 
+
+### Example
+Here is an example highlighting how a user can configure a route that forwards traffic to both a Service and a local
+unix domain socket.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: backend-svc
+spec:
+  ports:
+    - name: http
+      port: 3000
+      targetPort: 3000
+  selector:
+    app: backend
+---
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: Backend
+metadata:
+  name: backend-mixed-ip-uds
+spec:
+  applicationProtocol: H2C
+  addresses:
+    - type: UDS
+      unixDomainSocketAddress:
+        path: /var/run/backend.sock
+    - type: IPv4  
+      socketAddress:
+        address: 10.244.0.28
+        port: 3000
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: backend
+spec:
+  parentRefs:
+    - name: eg
+  hostnames:
+    - "www.example.com"
+  rules:
+    - backendRefs:
+        - group: gateway.envoyproxy.io
+          kind: Backend
+          name: backend-mixed-ip-uds
+          weight: 1
+        - group: ""
+          kind: Service
+          name: backend
+          port: 3000
+          weight: 1          
+      matches:
+        - path:
+            type: PathPrefix
+            value: /
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: BackendTLSPolicy
+metadata:
+  name: policy-btls
+spec:
+  targetRef:
+    group: gateway.envoyproxy.io
+    kind: Backend
+    name: backend-mixed-ip-uds
+  tls:
+    caCertRefs:
+      - name: backend-tls-checks-certificate
+        group: ''
+        kind: ConfigMap
+    hostname: example.com
+```
+
+## Design Decisions
+* All instances of `BackendObjectReference` in Envoy Gateway MAY support referencing the `Backend` kind.
+* In some cases, Envoy Gateway may reject references to a `Backend` resource. For example, a backend with a UDS address 
+  will be rejected on xRoute references. 
```

Comment:
for xRoute lets also disable `localhost` and `127.0.0.1`

Reply:
I have no objection to doing this. However, one can easily bypass this check by using a FQDN that resolves to 127.0.0.1. This is also true for services with FQDN endpoints. 

The real solution here is to use something like a network policy to block this sort of egress traffic from the proxy pod. However, network policies are not supported in every K8s distribution (e.g. kind). 

I think that that covering this in a security doc is probably the best approach. 

WDYT?

Reply:
for services with FQDN endpoints / endpointslices , the service is owned by the app dev (lower priviledge) and the endpointslice is owned by the admin (higher priviledge) , so if the admin wants to muck around, we can let them, but this is more around not allowing app devs to be able to access the proxy endpoint directly . I'd try to avoid as much as we can, and document what we can't in a separate doc

Reply:
Updated the design doc to reflect this. Will implement the validation in the translator, as it's specific to xRoutes and not all instances of `Backend`.